### DISCUSSION TITLE
Windows builds

### DISCUSSION THREAD

**Discussion by iAmPedestrian** [choria-io/general#41] [General]
hello

we are in need of the Windows binaries (maybe also the MSI) and since you are no longer providing them with the new versions I just want to confirm that the steps I'm doing are enough for the successful builds. I also can't use docker and the build is done on Windows 2022 server.

1. download/clone the source
2. set ENV variables: `GOOS=windows`, `GOARCH=amd64`
3. `go generate --run plugin`
4. `go build` with arguments and flags

when binary is compiled I have this buildinfo:
```
C:\tools\go-choria> .\choria-0.27.0-windows-amd64.exe buildinfo
Choria build settings:

Build Data:

     Version: 0.27.0
     Git SHA: daca201c
  Build Date: 2023-08-24 22:54:22 +02:00
     License: Apache-2.0
  Go Version: go1.21.0

Protocol Defaults and Settings:

     Default Collectives: mcollective
  Client Identity Suffix: mcollective

Network Broker Settings:

       Maximum Network Clients: 50000
  Embedded NATS Server Version: 2.9.21

Server Settings:

    Provisioning Target Resolver: Choria JWT Resolver
           Supports Provisioning: false
           Provisioning JWT file: C:\ProgramData\Choria\etc\provisioning.jwt

Security Defaults:

            TLS: true
  x509 Security: true

Agent Providers:

  External MCollective Agent Compatibility version 0.27.0
  Golang MCollective Agent Compatibility version 0.27.0
  Ruby MCollective Agent Compatibility version 0.27.0

Data Providers:

  Choria Data version 0.27.0
  Config_item Data version 0.27.0
  Machine_state Data version 0.27.0
  Scout Data version 0.27.0

Autonomous Agent Watchers:

  Archive Watcher version 0.27.0
  Exec Watcher version 0.27.0
  File Watcher version 0.27.0
  Gossip Watcher version 0.27.0
  Homekit Watcher version 0.27.0
  Kv Watcher version 0.27.0
  Metric Watcher version 0.27.0
  Nagios Watcher version 0.27.0
  Plugins Watcher version 0.27.0
  Schedule Watcher version 0.27.0
  Timer Watcher version 0.27.0
```
As said, I just want to confirm that this is enough and I'm not missing any steps.

---
Not really an issue, we can distribute the binary, but when I tried to build MSI with the WIX toolset and XML provided, I build the MSI and installed successfully, however, when I then tried to populate buildinfo on the binary I got this error:
```
C:\Program Files\Choria\bin>Choria.exe buildinfo
appbuilder setup failed: definition not found: Choria.exe
```
I tried build the MSI on several versions (0.25, 0.27 and also nightly) and also the Win10 machine, all with same result. Do you, by any chance, have the idea, what could be wrong there?

**Comment by ripienaar** [comment]
I think you're actually hitting a bug - you got the build right enough

https://github.com/choria-io/go-choria/blob/daca201c3778bdc93f619f22e60740c4fa171718/cmd/cmd.go#L63-L71

Here we check the invoked name to see if we should initialize AppBuilder (https://choria-io.github.io/appbuilder/) but this is quite a bad check.  Can you for example check if just running `choria` works, not `Choria.exe`?

**Reply by iAmPedestrian** [reply]
yes, `choria` and also `choria.exe` work and I got the information

seems that anything with lowercase `choria` in the name will work

**Reply by ripienaar** [reply]
Cheers, then I will add some `strings.ToLower()` there and fix the bug :)

**Comment by ripienaar** [comment]
Fixed in https://github.com/choria-io/go-choria/pull/2080

**Comment by iAmPedestrian** [comment]
For anyone interested, I've created the PowerShell script to build the binary and MSI with the steps outlined above
https://github.com/iAmPedestrian/choria-windows-build
